### PR TITLE
Port Array Union and Intersect to `functions-array`

### PR DESCRIPTION
**Pull request by jayzhan211**
## Which issue does this PR close?

Parts of #9285

## What changes are included in this PR?

move `array distinct` to `set_ops`

### CODE REVIEW

File: datafusion/functions-array/src/core.rs [hunk 1]
```diff
@@ -105,7 +105,8 @@ impl ScalarUDFImpl for MakeArray {
     }
 }
 
-/// `make_array` SQL function
+/// `make_array_inner` is the implementation of the `make_array` function.
+/// Can be used if the expected input and output are ArrayRef
 pub(crate) fn make_array_inner(arrays: &[ArrayRef]) -> Result<ArrayRef> {
```
Comment (jayzhan211) [inline datafusion/functions-array/src/core.rs:110]: rename to avoid conflict with function name

**Comment by Weijun-H** [comment]
I am considering moving `array-except` to setop too.

File: datafusion/functions-array/src/core.rs [hunk 2]
```diff
@@ -105,7 +105,8 @@ impl ScalarUDFImpl for MakeArray {
     }
 }
 
-/// `make_array` SQL function
+/// `make_array_inner` is the implementation of the `make_array` function.
+/// Can be used if the expected input and output are ArrayRef
```
Comment (comphead) [inline datafusion/functions-array/src/core.rs:109]:
```suggestion
```

**Review by comphead — APPROVED** [review]
lgtm thanks @jayzhan211

**Pull request merged**